src/clear/model/json_deserialize.cr

```diff
@@ -24,25 +24,26 @@ macro columns_to_instance_vars
     {% for name, settings in COLUMNS %}
       @[JSON::Field(presence: true)]
       getter {{name.id}} : {{settings[:type]}} {% unless settings[:type].resolve.nilable? %} | Nil {% end %}
+      
       @[JSON::Field(ignore: true)]
       getter? {{name.id}}_present : Bool
     {% end %}
 
     # Create a new empty model and fill the columns with object's instance variables
-    def create
-      assign_columns({{@type}}.new)
+    def create(permit, mass_assignment)
+      assign_columns({{@type}}.new, permit, mass_assignment)
     end
 
     # Update the inputted model and assign the columns with object's instance variables
-    def update(model)
-      assign_columns(model)
+    def update(model, permit, mass_assignment)
+      assign_columns(model, permit, mass_assignment)
     end
 
     macro finished
       # Assign properties to the model inputted with object's instance variables
-      protected def assign_columns(model)
+      protected def assign_columns(model, permit : Enumerable(String | Symbol), mass_assignment : Bool)
         {% for name, settings in COLUMNS %}
-          if self.{{name.id}}_present?
+          if self.{{name.id}}_present? && ((mass_assignment && permit.size == 0) ? true : (permit.any? { |p| p == {{name.stringify.id}} || p == {{name.id.symbolize}} }))
```

Comment:
this condition is quite confusing, and can be made simpler
not really sure what the purpose of the `mass_assignment` variable is here
```suggestion
           if self.{{name.id}}_present? && (!mass_assignment || permit.any? { |p| p == {{name.id.stringify}} || p == {{name.id.symbolize}} })
```